### DISCUSSION TITLE
What agent for target client

### DISCUSSION THREAD

**Discussion by Krapoh** [Security-Onion-Solutions/securityonion#10587] [2.4]
Hello,
I would like to know what agent I need to install on the target client/server that needs to get the data.
I saw the picture attached and found many agents supporting each thing and that made me confused.

![MicrosoftTeams-image](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/fd8a67ca-c24b-4e9d-a6d3-1421dac818ad)

Please let me know if should I install all the APIs during installation.
Thank you.

**Comment by dougburks** [comment]
That Data Types graphic was made for Security Onion 2.3 where you could choose from several different endpoint agents:
https://docs.securityonion.net/en/2.3/host.html

In Security Onion 2.4, you'll want to use the Elastic Agent:
https://docs.securityonion.net/en/2.4/host.html

**Reply by Krapoh** [reply]
The Elastic Agent is not an installer program but a batch file to run to get information and sent data back to Kibana.
Don't you have an installer file that makes it always run on the target server?

By the way, Kibana got data from the target server as expected after running the batch file.
Thank you.

**Reply by dougburks** [reply]
Did you download the Elastic Agent as shown in the documention?
https://docs.securityonion.net/en/2.4/elastic-agent.html#elastic-agent

It should create a service on the endpoint.

**Comment by Krapoh** [comment]
![MicrosoftTeams-image](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/0c1e38ac-8106-4c69-9f7e-10ed5a59b739)


Dougburks, I have not found the installer you provide.
pls, advise which one I should download.

**Reply by dougburks** [reply]
Since you created this discussion in the 2.4 category, I assumed you were using Security Onion 2.4. However, your screenshot shows that you're using version 2.3.250. Since you're using 2.3, you'll want to choose one of the options on the Downloads page shown in your screenshot. For more information, see the 2.3 link that I provided earlier:
https://docs.securityonion.net/en/2.3/host.html

**Reply by Krapoh** [reply]
Sorry, for the late reply, I did install version 2.4 and installed the agent on the target server but the status keeps "Unhealthy".
Then I restarted the services on the server and the status is back to "Healthy" and repeat to "Unhealthy" again.
Would you know the possible cause of this issue?  pls, advise.

thank you.


![Screenshot_1](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/5d32e7e2-8bde-4912-8127-b3587ddec92b)


![Screenshot_2](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/4065475b-ca97-4289-ab7e-7664accc7db6)
